frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -256,6 +278,32 @@ pub mod pallet {
 
             T::DbWeight::get().writes(5)
         }
+
+        fn on_runtime_upgrade() -> Weight {
```

Comment:
Better don't this feature, let it be external callable function. Parity plan to remove `on_runtime_upgrade` hook soon.

Reply:
So you mean that I just expose it as a callable and call it directly from runtime?
I'm looking for an example of an alternative way but couldn't find it.

Would it be better to wait until the hooks is removed from the code so we can see how
other Parity pallets will be updated?
I'm afraid we could end up implementing our own solution now and then later adapting it
to the new solution introduced by Parity. Possible double work.

Reply:
Parity suggest to call pallet migrations from top-level upgrade handler, like here https://github.com/paritytech/cumulus/blob/polkadot-v0.9.11/polkadot-parachains/statemine/src/lib.rs#L777

Reply:
I see, I think I misunderstood you the first time.

Updates are pushed, please let me know if it looks ok.
I will test them again now.

Reply:
@akru are you ok with the storage upgrade code after the latest change?

Reply:
No problem, it's ok